Wrapping/Generators/Python/Tests/extras.py

```diff
@@ -464,11 +474,50 @@ def custom_callback(name, progress):
     except ValueError:
         pass
 
+    # Check empty array
     empty_array = np.array([], dtype=np.uint8)
     empty_array.shape = (0, 0, 0)
     empty_image = itk.image_from_array(empty_array)
     empty_da = itk.xarray_from_image(empty_image)
     empty_image_round = itk.image_from_xarray(empty_da)
+
+    # Check order
+    arr = np.random.randint(0, 255, size=(4,5,6), dtype=np.uint8)
+    data_array = xr.DataArray(arr, dims=['z','y','x'])
+    image = itk.image_from_xarray(data_array)
+    assert np.allclose(arr, itk.array_view_from_image(image))
+    assert np.allclose(arr.shape, itk.array_view_from_image(image).shape)
+
+    data_array = xr.DataArray(arr, dims=['x','y','z'])
+    image = itk.image_from_xarray(data_array)
+    assert np.allclose(arr.transpose(), itk.array_view_from_image(image))
+    assert np.allclose(arr.shape[::-1], itk.array_view_from_image(image).shape)
+
+    data_array = xr.DataArray(arr, dims=['y','x','c'])
+    image = itk.image_from_xarray(data_array)
+    assert np.allclose(arr, itk.array_view_from_image(image))
+    assert np.allclose(arr.shape, itk.array_view_from_image(image).shape)
+
+    data_array = xr.DataArray(arr, dims=['c','x','y'])
+    image = itk.image_from_xarray(data_array)
+    assert np.allclose(arr.transpose(), itk.array_view_from_image(image))
+    assert np.allclose(arr.shape[::-1], itk.array_view_from_image(image).shape)
+
+    data_array = xr.DataArray(arr, dims=['q','x','y'])
+    try:
+        image = itk.image_from_xarray(data_array)
+        assert False
+    except ValueError:
+        pass
+
+    if '(<itkCType bool>, 4)' in itk.Image.GetTypesAsList():
+        arr = np.random.randint(0, 255, size=(4,5,6,3), dtype=np.uint8)
```

Comment:
If it was an issue above then here too: `<itkCType bool>` and `np.uint8`.

Reply:
Good catch. Addressed.